sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/Release.java

```diff
@@ -346,6 +371,10 @@ public boolean equals(Object o) {
     return artifact.equals(release.artifact) && version.equals(release.version);
   }
 
+  public void setProduct(Product product) {
+    this.product = product;
+  }
+
   @Override
   public int hashCode() {
```

Comment:
Probably the hashCode method has to be modified.
I am not sure if it is a valid use case, but I saw that it is not possible to add the same version for two different products.
It fails add the addRelease phase due to the equality of two releases with same version and different product.

Reply:
Same for the to String.
Should the product be added there?

Reply:
We don't need to support this usecase but I made these fixes.